data_structures/src/chain.rs

```diff
@@ -50,17 +50,24 @@ pub struct ConsensusConstants {
     pub reputation_punishment: f64,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
-pub struct Checkpoint {
-    pub number: u32,
-    pub hash: Vec<u8>,
+/// Checkpoint beacon structure
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct CheckpointBeacon {
+    /// The serial number for an epoch
+    pub checkpoint: Epoch,
+    /// The 256-bit hash of the previous block header
+    pub hash_prev_block: Vec<u8>,
 }
 
-impl Default for Checkpoint {
-    fn default() -> Checkpoint {
-        Checkpoint {
-            number: 0,
-            hash: Vec::new(),
+impl Default for CheckpointBeacon {
+    fn default() -> CheckpointBeacon {
+        CheckpointBeacon {
+            checkpoint: Epoch(0),
+            hash_prev_block: Vec::new(),
```

Comment:
I like this more, but anyway we will need to change it in the future to account generalization over different hash functions. @mariocao is working on it as part of #164 .
```suggestion
            hash_prev_block: vec![0; 32],
```